cmd/up.go

```diff
@@ -36,55 +31,24 @@ func init() {
 }
 
 func runCmdUp(cmd *cobra.Command, args []string) error {
-	// Up flags.
-	required := []struct {
-		name, val string
-	}{
-		{"--s3-uri", upOpts.s3URI},
-	}
-	var missing []string
-	for _, req := range required {
-		if req.val == "" {
-			missing = append(missing, strconv.Quote(req.name))
-		}
-	}
-	if len(missing) != 0 {
-		return fmt.Errorf("Missing required flag(s): %s", strings.Join(missing, ", "))
-	}
-
-	conf, err := config.ClusterFromFile(configPath)
-	if err != nil {
-		return fmt.Errorf("Failed to read cluster config: %v", err)
-	}
-
-	opts := stackTemplateOptions(upOpts.s3URI, upOpts.prettyPrint)
-
-	cluster, err := cluster.NewCluster(conf, opts, upOpts.awsDebug)
-	if err != nil {
-		return fmt.Errorf("Failed to initialize cluster driver : %v", cluster)
-	}
-
-	if err := cluster.ValidateUserData(); err != nil {
+	if err := validateRequired(flag{"--s3-uri", upOpts.s3URI}); err != nil {
```

Comment:
Thanks for the question but no!
s3URI is required in order to render stack templates because the URI is parsed and then included in the stack templates as (1) URLs to actual worker/controller cloud-configs in S3 and (2) URLs to nested stack templates referenced from the root stack template.